libraries/standard/mqtt/src/private/iot_mqtt_helper.h

```diff
@@ -277,7 +277,32 @@ IotMqttError_t _IotMqtt_ProcessIncomingPublishFlags( uint8_t publishFlags,
  *
  * @return Pointer to the end of the encoded string, which will be identical to
  * `pDestination` if nothing was encoded.
+ *
+ * @warning This function does not check the size of `pDestination`! To avoid a buffer
```

Comment:
The caller appears to have this value, though. Why don't we add a parameter to the below to take it in, then add a check for it?

Reply:
This warning is a sync from the changes of #761 to functions that were duplicated as part of #772. This was the [reasoning](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/761/#discussion_r374288528) for not adding a similar check right now, and to add the [warning](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/761/#discussion_r374293230) instead. The buffer size is checked before `_IotMqtt_SerializeConnectCommon` is called, which calls this function after others that also write into the buffer. Checking the buffer a second time only for this function seems out of place

Reply:
Let's defer this for a future revision since it's outside of the scope of this PR and a larger topic.